### PR TITLE
TransactionBroadcast: internally create/complete `sentFuture`

### PR DESCRIPTION
**Pull request by msgilligan**
* Add a private member called `sentFuture` that completes when the broadcast is "sent" (written to buffer) to the broadcast peers
* Update the `broadcast()` method to track success/failure of the individual broadcast messages and update `sentFuture` when the individual messages complete. (This is done by using `thenComposeAsync`/`whenComplete`)
* Add comments and TODOs

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/TransactionBroadcast.java [hunk 1]
```diff
@@ -130,11 +134,30 @@ public Message onPreMessageReceived(Peer peer, Message m) {
...
+     *     <li>Broadcast the transaction by a determined number of {@link org.bitcoinj.core.Peer}s</li>
+     *     <li>Wait for confirmation from a determined number of remote peers that they have received the broadcast</li>
+     *     <li>Mark {@link TransactionBroadcast#future()} ("sent future") as complete</li>
+     * </ol>
+     * <p>
+     * It should further be noted that "broadcast" in this class means that
+     * {@link org.bitcoinj.net.MessageWriteTarget#writeBytes} has completed successfully which means the message has
+     * been sent to the "OS network buffer" -- see {@link org.bitcoinj.net.MessageWriteTarget#writeBytes} or its implementation.
+     * <p>
+     * TODO: Should this method be moved into the PeerGroup?
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/TransactionBroadcast.java:153]: I think this TODO should not go into the JavaDoc.

**Review by schildbach — COMMENTED** [review]
Looks great!

File: core/src/main/java/org/bitcoinj/core/TransactionBroadcast.java (hunk 1 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/TransactionBroadcast.java:153]: OK, I'll fix this.

**Comment by msgilligan** [comment]
This commit https://github.com/bitcoinj/bitcoinj/pull/2584/commits/3a6a2a2d09a02d5243d5f10aeb4de6e591d4c4d4 makes the new `sentFuture` available.

**Comment by schildbach** [comment]
Merged!

**Pull request merged**